### ISSUE TITLE
if barCategoryGap and barSize are incompatible, how can I choose the barGap?

### ISSUE THREAD

**Issue by fanysalgadomrf** [recharts/recharts#3680]
```
<ComposedChart
	data={ data.data }
	barCategoryGap={ 1 }
        barSize={ 1 }
>
<Bar
	radius={ [4, 4, 0, 0] }
	dataKey="value"
	yAxisId="left"
>
	<Cell .../>
</Bar>
....
</ComposedChart>
```
If barCategoryGap and barSize are incompatible, how can I choose the size of the bars and paint 1-pixel wide bars? I only have the barCategoryGap property in the composite chart, but I want to paint 1-pixel wide bars close together. What can I do to remove this gap? I want to paint the bars close together.
<img width="451" alt="Captura de Pantalla 2023-08-02 a las 11 05 05" src="https://github.com/recharts/recharts/assets/77836336/94c4f773-7e82-449f-9f32-94cf7335a095">

In this second example, I included barSize but I couldn't change the gap. I want to paint thin bars very close (without a gap).
```
<ComposedChart
....
      barCategoryGap={ 1 }
      barSize={ 1 }
>...</ComposedChart>
```
<img width="453" alt="Captura de Pantalla 2023-08-02 a las 11 07 14" src="https://github.com/recharts/recharts/assets/77836336/d4aee8cb-895d-4268-8e19-9fd5f340f52a">

**Comment by ckifer** [comment]
There is also `maxBarSize` and `barGap`, can you try using those to achieve the desired result?

https://codesandbox.io/s/simple-bar-chart-forked-pqtynh?file=/src/App.tsx

I'm not sure its going to be possible to have the bars touching and for them to be 1px. In order for that to happen you'll need a lot more data items. The `XAxis` is plotted evenly and will space things out based on the data its given. 1px bars and no space between them doesn't give very well to even spacing.

**Comment by fanysalgadomrf** [comment]
Thank you so much for your idea, Ckifer. My chart has 90 bars. In little screens (mobiles for example) the bars are cut. For this reason, I want to try to reduce the bar size and close the bars but I didn't find a solution :(
I am exploring other ways to solve my problem. If I use width in the chart container, the bars are closer and don't disappear. I don't like this solution because I have to use many media queries :( but I didn't find the solution.

**Comment by fanysalgadomrf** [comment]
> There is also `maxBarSize` and `barGap`, can you try using those to achieve the desired result?
> 
> https://codesandbox.io/s/simple-bar-chart-forked-pqtynh?file=/src/App.tsx
> 
> I'm not sure its going to be possible to have the bars touching and for them to be 1px. In order for that to happen you'll need a lot more data items. The `XAxis` is plotted evenly and will space things out based on the data its given. 1px bars and no space between them doesn't give very well to even spacing.

[fanysalgadomrf](https://github.com/fanysalgadomrf) commented [now](https://github.com/recharts/recharts/issues/3680#issuecomment-1665089399)
Thank you so much for your idea, Ckifer. My chart has 90 bars. In little screens (mobiles for example) the bars are cut. For this reason, I want to try to reduce the bar size and close the bars but I didn't find a solution :(
I am exploring other ways to solve my problem. If I use width in the chart container, the bars are closer and don't disappear. I don't like this solution because I have to use many media queries :( but I didn't find the solution.

**Comment by ckifer** [comment]
Hmm, responsive container should help scale your bar size down without much other issue. What do you mean the bars are cut?

**Comment by fanysalgadomrf** [comment]
> Hmm, responsive container should help scale your bar size down without much other issue. What do you mean the bars are cut?

Hi ckifer, I finally found the real problem. My graph is inside a table and that was the problem. I don't know why but the table was affecting the responsive container and it was not working properly. I took my graph out of the table and my chart and the responsive container worked without any problem. ⬇️ 

<img width="510" alt="Captura de Pantalla 2023-08-08 a las 9 16 38" src="https://github.com/recharts/recharts/assets/77836336/c5a105d7-c426-4491-8c17-10bedb8f07a5">

Thank you for your interest and help 🙏🏻 . Now I have to fight with the table 😅
